core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/listener/DAnnotationPrecommitListener.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2021 THALES GLOBAL SERVICES. All rights reserved. 
+ *******************************************************************************/
+package org.polarsys.capella.core.sirius.ui.listener;
+
+import org.eclipse.emf.common.notify.Notification;
+import org.eclipse.sirius.business.api.session.Session;
+import org.eclipse.sirius.business.api.session.SessionManagerListener;
+import org.eclipse.sirius.business.internal.session.SessionEventBrokerImpl;
+import org.eclipse.sirius.viewpoint.DRepresentation;
+import org.eclipse.sirius.viewpoint.DRepresentationDescriptor;
+import org.eclipse.sirius.viewpoint.description.DAnnotation;
+import org.eclipse.sirius.viewpoint.description.Viewpoint;
+
+import com.google.common.base.Predicate;
+
+/**
+ * A {@link SessionManagerListener} listening to notification concerning DAnnotations and triggering
+ * {@link DAnnotationChangeTrigger}.
+ * 
+ * @author <a href="mailto:steve.monnier@obeo.fr">Steve Monnier</a>
+ */
+public class DAnnotationPrecommitListener implements SessionManagerListener {
+
+    private DAnnotationChangeTrigger dAnnotationChangeTrigger;
+
+    Predicate<Notification> considerDAnnotationForAutomaticRefreshPredicate = notification -> {
+        if (notification != null) {
+            if (notification.getNotifier() instanceof DAnnotation) {
+                // A DAnnotation has been updated
```

Comment:
as the GMF_Diagram is also stored on DRepresentation, can you test that create a new diagram doesnt refresh it twice or if the edition of the GMF diagram doesn't trigger unexpected refreshes

Reply:
I will look into it, but that seems like a different topic.

Reply:
I updated the test by adding a new ChangeTrigger that counts how much refresh are asked.